src/frontends/tensorflow_common/src/op/reciprocal.cpp

```diff
@@ -16,12 +17,37 @@ namespace op {
 
 OutputVector translate_reciprocal_op(const NodeContext& node) {
     // computes element-wise 1/x, where x - input
-    default_op_checks(node, 1, {"Reciprocal"});
+    default_op_checks(node, 1, {"Reciprocal"}, true);
     auto x = node.get_input(0);
-    auto minus_one_const = create_same_type_const_scalar<int32_t>(x, -1);
-    auto reciprocal = make_shared<v1::Power>(x, minus_one_const);
-    set_node_name(node.get_name(), reciprocal);
-    return {reciprocal};
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+
+    if (complex_type_mark_x) {
+        x = complex_type_mark_x->input_value(0);
+        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto gather_index_imag = make_shared<v0::Constant>(element::i32, Shape{}, 1);
```

Comment:
```suggestion
        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
        auto gather_index_imag = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
```
I think we do not need to loose auxiliary dimension, otherwise, we will not be able to concatenate back below.